app/services/report_forwarding_service.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class ReportForwardingService < BaseService
+  include Payloadable
+
+  def call(report, forwarder, options = {})
+    @report = report
+    @options = options
+    @forwarded_to = []
+
+    return unless @report.forwardable?
+
+    unless forward_to_domains.empty?
+      forward_to_origin! if forward_to_origin?
+      forward_to_replied_to!
+    end
+
+    unless @forwarded_to.empty?
+      report.update!({
+        forwarded_at: DateTime.now.utc,
+        forwarded_by: forwarder,
+        forwarded_to_domains: @forwarded_to,
+      })
+    end
```

Comment:
There's probably a better way to write this, but the idea is that the report starts out as a record that's not yet forwarded, since `ReportService` doesn't do the forwarding itself, and then `ReportForwardingService` does the forward and updates the report to reflect the new state.